File: diploma-server/src/main/java/org/arhor/diploma/domain/Updatable.java
package org.arhor.diploma.domain;

import java.time.temporal.TemporalAccessor;

@SuppressWarnings({"unused"})
public interface Updatable<T extends TemporalAccessor> {

  T getUpdated();

  void setUpdated(T updated);

  default void onUpdate() { /* do nothing by default */ }
}

File: diploma-server/src/main/kotlin/org/arhor/diploma/domain/Updatable.kt
package org.arhor.diploma.domain

import java.time.temporal.TemporalAccessor

interface Updatable<T : TemporalAccessor> {

  fun getUpdated(): T?

  fun setUpdated(updated: T?)

  @JvmDefault
  fun onUpdate() { /* do nothing by default */ }
}
